GLOSSARY.md

```diff
@@ -0,0 +1,51 @@
+# Parts of Class Names
+
+| Name      | Meaning                                                                                                                                                                                                        |
+| --------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Abstract* | An incomplete class used for *dynamic* polymorphism. It has pure virtual methods that need to be implemented by subclasses.                                                                                    |
+| Base*     | An incomplete class used for *static* polymorphism. It implements methods shared by all implementations. Mostly, it is used so that we have a non-templated superclass, which can be stored in e.g., a vector. |
```

Comment:
I'm not sure what you mean by "static polymorphism", which to me is overloading methods, templates and so forth, in the context of this prefix.

Reply:
Does this help:

| Name      | Meaning                                                                                                                                                                                                                                                |
| --------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| Abstract* | An incomplete class used for *dynamic* polymorphism (i.e., OOP with run-time resolved virtual methods). It has pure virtual methods that need to be implemented by subclasses.                                                                         |
| Base*     | An incomplete class used for *static* polymorphism (i.e., compile-time resolved templates). It implements methods shared by all implementations. Mostly, it is used so that we have a non-templated superclass, which can be stored in e.g., a vector. |